src/uct/ib/ud/base/ud_ep.c

```diff
@@ -989,12 +1083,19 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         if ((size_t)byte_len == sizeof(*neth)) {
             goto out;
         }
+
         if (neth->packet_type & UCT_UD_PACKET_FLAG_CTL) {
-            uct_ud_ep_rx_ctl(iface, ep, neth, skb);
-            goto out;
+            uct_ud_ep_rx_ctl(iface, ep, neth);
+        } else if (neth->packet_type & UCT_UD_PACKET_FLAG_PUT) {
+            /* TODO: remove once ucp implements put */
+            uct_ud_ep_rx_put(neth, byte_len);
+        } else {
```

Comment:
why needed to refactor this?

Reply:
moved handling PUT back.
it was refactored to put CREQ/CREP/FIN to frag list too - to allow sending CREQ/FIN using previous CREQ/CREP code at the middle of the endpoint's life .